src/main/java/com/JellyWorks/Utility/PreprocessFile.java

```diff
@@ -0,0 +1,109 @@
+package com.JellyWorks.Utility;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import com.JellyWorks.Utility.TypeChecks;
+import com.JellyWorks.storage.redis.RedisProcessor;
+
+import org.springframework.cache.annotation.Cacheable;
+import java.net.URISyntaxException;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.data.redis.core.ValueOperations;
+
+public class PreprocessFile {
+
+	private String path;
+	private InputStream inputStream;
+	private ConfigurableApplicationContext configurableApplicationContext;
+	
+	public PreprocessFile(InputStream i) throws IOException{
+		this.inputStream=i;
+		this.path="C:/Users/Ramya/Desktop/AWS/";
+		this.configurableApplicationContext = new AnnotationConfigApplicationContext(RedisProcessor.class);
+	}
+	@Cacheable("splitFile")
+	public boolean splitFile() throws URISyntaxException, Exception {
+		BufferedReader reader=new BufferedReader(new InputStreamReader(inputStream));
+		BufferedWriter writer=null;
+		try{
+			String line="";
+			int curr_month=0;
+			String out_folder="Output";

```

Comment:
such variables can be made final... Usually we use a checkstyle that makes such suggestions during build. This is called static code analysis. Read more here: http://continuousdev.com/2015/08/checkstyle-vs-pmd-vs-findbugs/